### PR TITLE
WIP: Bsweger/add tree class/28

### LINKED ISSUE #28: Add a CladeTime method to return an "as_of" reference tree
To simplify the process of creating target data, add a method to `CladeTime` that gets a reference tree based on its `tree_as_of` attribute.

**Definition of Done**

Once complete, users should be able to do something like this:

```python
from virus_clade_utils.cladetime import CladeTime

ct = CladeTime(sequence_as_of="2024-09-01", tree_as_of="2024-07-31")

# get reference tree that was in use as of 2024-07-31:
ct.reference_tree()
```

### PR DESCRIPTION
**Pull request by bsweger**
Resolves #28 

This PR adds a `Tree` class to cladetime. `Tree` has two properties: url to a reference tree, and the tree itself.

The changes also include improvements to the existing documentation, as well as new documentation for `Tree`.

### Note to reviewer(s)

A great way to test the new features, would be to give [the Tree documentation example](https://cladetime--39.org.readthedocs.build/en/39/user-guide.html#reference-trees) a spin.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,289 +1,117 @@
...
+# SARS-CoV-2 past sequence data and metadata files
+>>> ct.url_sequence
+'https://nextstrain-data.s3.amazonaws.com/files/ncov/open/sequences.fasta.zst?versionId=8Zszokay3LRP5Zec_cviQ8oXkx8cJlwq'
+>>> ct.url_sequence_metadata
+'https://nextstrain-data.s3.amazonaws.com/files/ncov/open/metadata.tsv.zst?versionId=U4aIlh5HI1XuDLPW7q9WTZad6gXwARqT'
 ```
 
+### Interacting with sequence metadata
 
-## Docker Setup
```
Comment (bsweger) [inline README.md:108]: Opened an issue to [move the setup instructions to a new CONTRIBUTING.md file](https://github.com/reichlab/cladetime/issues/40).

At this point, the dev setup distracts from the main focus of the page, which is how to get started.

File: docs/user-guide.rst
Comment (bsweger) [inline docs/user-guide.rst:1]: Made some heavy edits to better split up the examples.

File: src/cladetime/cladetime.py [hunk 2]
```diff
@@ -150,7 +149,7 @@ def sequence_metadata(self):
     @sequence_metadata.getter
     def sequence_metadata(self) -> pl.LazyFrame:
         """
-        :class:`polars.LazyFrame` : A Polars LazyFrame that references
+        :external+polars:std:doc:`polars.LazyFrame<reference/lazyframe/index>` : A Polars LazyFrame that references
```
Comment (bsweger) [inline src/cladetime/cladetime.py:152]: _Finally_ figured out how to link to documentation for other projects that use sphinx

File: src/cladetime/util/session.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ def _get_session(retry: bool = True) -> requests.Session:
             total=5,
             allowed_methods=frozenset(["GET", "POST"]),
             backoff_factor=1,
-            status_forcelist=[401, 403, 404, 429, 500, 502, 503, 504],
+            status_forcelist=[429, 500, 502, 503, 504],
```
Comment (bsweger) [inline src/cladetime/util/session.py:28]: There's no reason to retry 401, 403, and 404 status codes since we don't expect the result to change.

**Review by elray1 — APPROVED** [review]
approved via live walk-through

**Pull request merged**